packages/dashboard/src/components/widgets/dynamicWidget.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { TimeQuery, TimeSeriesData, TimeSeriesDataRequest } from '@iot-app-kit/core';
+import {
+  Annotations,
+  LabelsConfig,
+  Trend,
+  MovementConfig,
+  ScaleConfig,
+  LayoutConfig,
+  Axis,
+  LegendConfig,
+  AlarmsConfig,
+  MessageOverrides,
+  MinimalSizeConfig,
+  MinimalViewPortConfig,
+} from '@synchro-charts/core';
+
+import {
+  BarChart,
+  Kpi,
+  LineChart,
+  ResourceExplorer,
+  ScatterChart,
+  StatusGrid,
+  StatusTimeline,
+  Table,
+} from '@iot-app-kit/react-components';
+
+import './dynamicWidget.css';
+import { ComponentTag } from '../../types';
+
+// eslint-disable-next-line
+const IconX = require('./iconx.svg') as string;
+
+// eslint-disable-next-line
+export const ComponentMap: { [key in ComponentTag]: any } = {
+  'iot-bar-chart': BarChart,
+  'iot-kpi': Kpi,
+  'iot-line-chart': LineChart,
+  'iot-resource-explorer': ResourceExplorer,
+  'iot-scatter-chart': ScatterChart,
+  'iot-status-grid': StatusGrid,
+  'iot-status-timeline': StatusTimeline,
+  'iot-table': Table,
+};
+
+export type DynamicWidgetProps = {
+  componentTag: ComponentTag;
+  widgetId: string;
+  queries: TimeQuery<TimeSeriesData[], TimeSeriesDataRequest>[];
+  viewport: MinimalViewPortConfig;
+  movement?: MovementConfig;
+  scale?: ScaleConfig;
+  layout?: LayoutConfig;
+  legend?: LegendConfig;
+  annotations?: Annotations;
+  axis?: Axis.Options;
+  messageOverrides?: MessageOverrides;
+  size?: MinimalSizeConfig;
+  trends?: Trend[];
+  alarms?: AlarmsConfig;
+  gestures?: boolean;
+  labelsConfig?: LabelsConfig;
+  readOnly?: boolean;
+  isEditing?: boolean;
+  invalidTagErrorHeader: string;
+  invalidTagErrorSubheader: string;
+};
+
+const DynamicWidgetComponent: React.FC<DynamicWidgetProps> = ({
+  componentTag,
+  widgetId,
+  queries,
+  viewport,
+  movement,
+  scale,
+  layout,
+  legend,
+  annotations,
+  axis,
+  messageOverrides,
+  size,
+  trends,
+  alarms,
+  gestures,
+  labelsConfig,
+  readOnly,
+  isEditing,
+  invalidTagErrorHeader,
+  invalidTagErrorSubheader,
+}) => {
+  const Component = ComponentMap[componentTag];
+  const componentIsRegistered = typeof Component !== 'undefined';
+
+  // eslint-disable-next-line
+  const props: any = {
```

Comment:
Non blocking: Is this the place where you were having difficulties with incompatible types across components? Do we think this is an intractable problem, or are we punting on it?

Reply:
Punting on it for now. It will be a little easier to solve when we move the package to react-components since that package already depends on components. It was a mess to depend on core / components / react-components to put together a single type in dashboard to be passed into react-components. We might still have to create some type in the components package, but we can start with a definition in react-components where those wrappers are defined.